File: backend/slotify/spring/src/main/java/com/slotify/backend/spring/company/dtos/UpdateIntervalRequest.java
package com.slotify.backend.spring.company.dtos;

import jakarta.validation.constraints.Min;
import jakarta.validation.constraints.NotNull;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;

import java.time.LocalDateTime;

@Getter
@Setter
@NoArgsConstructor
@ToString
public class UpdateIntervalRequest {

    @NotNull
    private LocalDateTime startDateTime;

    @NotNull
    private LocalDateTime endDateTime;

    @Min(0)
    @NotNull
    private Integer maxConcurrentService;
}

File: frontend/app/src/main/java/com/raj/slotify/dtos/company/UpdateIntervalRequest.kt
package com.raj.slotify.dtos.company

import com.google.gson.annotations.SerializedName
import java.time.LocalDateTime

data class UpdateIntervalRequest(

    @SerializedName("startDateTime")
    val startDateTime: LocalDateTime,

    @SerializedName("endDateTime")
    val endDateTime: LocalDateTime,

    @SerializedName("maxConcurrentService")
    val maxConcurrentService: Int

)
